### PR TITLE
Mots 346: Added Owner of locations.

### PR DESCRIPTION
**Pull request by pstudzinski**
Added Owner of locations. Fixed rights and tests.
https://applab.atlassian.net/browse/MOTS-346

### CODE REVIEW

File: src/main/java/org/motechproject/mots/errorhandling/GlobalErrorHandling.java [hunk 1]
```diff
@@ -57,4 +58,11 @@ public ErrorResponse entityNotFoundException(EntityNotFoundException ex) {
   public String handleAccessDeniedException(Exception ex, WebRequest request) {
     return ex.getMessage();
   }
+
+  @ExceptionHandler(LocationException.class)
```
Comment (pmuchowski) [inline src/main/java/org/motechproject/mots/errorhandling/GlobalErrorHandling.java]: I think you should rename this exception to something like this "MotsAccessDeniedException" or something similar, but indicating that the user don't have rights to access the resource

File: src/main/java/org/motechproject/mots/service/LocationService.java [hunk 2]
```diff
@@ -115,22 +115,69 @@ public Community createCommunity(Community community) {
...
+  }
+
+  @PreAuthorize(RoleNames.HAS_CREATE_FACILITIES_ROLE)
+  public Community createCommunity(Community community) {
+    return communityRepository.save(community);
+  }
+
+  /**
+   * Update Facility.
+   *
```
Comment (pmuchowski) [inline src/main/java/org/motechproject/mots/service/LocationService.java]: remove this empty line

File: src/main/java/org/motechproject/mots/service/LocationService.java [hunk 3]
```diff
@@ -115,22 +115,69 @@ public Community createCommunity(Community community) {
...
+   * @param facility facility to update
+   *
```
Comment (pmuchowski) [inline src/main/java/org/motechproject/mots/service/LocationService.java]: as above

File: src/main/java/org/motechproject/mots/service/LocationService.java [hunk 4]
```diff
@@ -115,22 +115,69 @@ public Community createCommunity(Community community) {
...
+      throw new LocationException("Could not edit facility, because you are not the owner");
+    }
+
     return facilityRepository.save(facility);
   }
 
-  @PreAuthorize(RoleNames.HAS_MANAGE_FACILITIES_ROLE)
+  /**
+   * Update Community.
+   *
```
Comment (pmuchowski) [inline src/main/java/org/motechproject/mots/service/LocationService.java]: as above

File: src/main/java/org/motechproject/mots/service/LocationService.java [hunk 5]
```diff
@@ -115,22 +115,69 @@ public Community createCommunity(Community community) {
...
+   * @param community community to update
+   *
```
Comment (pmuchowski) [inline src/main/java/org/motechproject/mots/service/LocationService.java]: as above

File: src/webapp/js/components/community-edit.js [hunk 6]
```diff
@@ -27,7 +27,8 @@ class CommunityEdit extends Component {
   }
 
   componentWillMount() {
-    if (!hasAuthority(MANAGE_USERS_AUTHORITY)) {
+    if (!hasAuthority(MANAGE_FACILITIES_AUTHORITY)
```
Comment (pmuchowski) [inline src/webapp/js/components/community-edit.js]: you can provide a list of rights to hasAuthority

File: src/webapp/js/components/facility-edit.js [hunk 7]
```diff
@@ -27,7 +27,8 @@ class FacilityEdit extends Component {
   }
 
   componentWillMount() {
-    if (!hasAuthority(MANAGE_USERS_AUTHORITY)) {
+    if (!hasAuthority(MANAGE_FACILITIES_AUTHORITY)
```
Comment (pmuchowski) [inline src/webapp/js/components/facility-edit.js]: as above

File: src/webapp/js/utils/authorization.js [hunk 8]
```diff
@@ -19,3 +21,14 @@ export function hasAuthority(...authorities) {
   const currentUserAuthorities = tokenDecoded.authorities;
   return _.some(currentUserAuthorities, el => _.includes(authorities, el));
 }
+
+export function canEditLocation(username) {
```
Comment (pmuchowski) [inline src/webapp/js/utils/authorization.js]: pass cellInfo.original to this method and then rename the parameter to location

**Comment by pstudzinski** [comment]
Naming Refactor after review.

**Pull request merged**